google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/flexible/AppEngineFlexibleStage.java

```diff
@@ -111,10 +111,16 @@ public boolean stage(@NotNull Path stagingDirectory) throws IOException {
         }
       }
 
-      Path stagedArtifactPath =
-          stagingDirectory.resolve(
-              "target"
-                  + AppEngineFlexibleDeploymentArtifactType.typeForPath(deploymentArtifactPath));
+      String stagedArtifactName = deploymentConfiguration.getStagedArtifactName();
+      if (Strings.isNullOrEmpty(stagedArtifactName)) {
+        stagedArtifactName =
```

Comment:
is this right? shouldn't the stagedArtifactName be the deployment source artifact name if this value is empty?